### PR TITLE
Fix bug CameraView on iOS: camera preview is not always rotated correct and photos are rotated

### PR DESCRIPTION
**Pull request by Forestbrook**
### Description of Change ###

 The Camera preview was not always correct because UIDevice.CurrentDevice.Orientation does not always return the correct rotation.

The photos were shown rotated, because the VideoOrientation of the AVCapturePhotoOutput connection was not set when taking a picture.

 ### Linked Issues ###

#2311 [BUG] CameraView on iOS: camera preview is not always rotated correct and photos are rotated

Fixes:
 - Observe OrientationDidChange notification from UIDevice to make sure the camera preview is always rotated correctly
 - Set VideoOrientation of the AVCapturePhotoOutput connection before taking the picture to make sure the photo will not be  rotated.
 - Optimized retrieving the image data after taking the picture.
 - On iOS 13+ used UIWindowScene from UIApplication.SharedApplication.ConnectedScenes to retrieve the correct video orientation.

Could not add tests because CameraManager.macios.cs is internal and the changes are iOS specific.

 ### PR Checklist ###

 - [ ] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [ ] Rebased on top of `main` at time of PR
 - [ ] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls

### CODE REVIEW

**Comment by Forestbrook** [comment]
I agree with the CLA.  I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.

> @Forestbrook please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @dotnet-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @dotnet-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @dotnet-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

File: src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs [hunk 1]
```diff
@@ -191,42 +196,44 @@ protected virtual async partial ValueTask PlatformTakePicture(CancellationToken
...
+			imageData = result.Photo.FileDataRepresentation?.AsStream();
+		}
+		catch
+		{
+			// possible exception: ObjCException NSInvalidArgumentException NSAllocateMemoryPages(...) failed in AVCapturePhoto.get_FileDataRepresentation()
+			imageData = null;
 		}
-		catch (Exception ex)
+
+		if (imageData == null)
```
Comment (bijington) [inline src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs:230]: We prefer to use `is` over `==`

```suggestion
		if (imageData is null)
```

File: src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs [hunk 2]
```diff
@@ -191,42 +196,44 @@ protected virtual async partial ValueTask PlatformTakePicture(CancellationToken
 
 		var capturePhotoSettings = AVCapturePhotoSettings.FromFormat(codecSettings);
 		capturePhotoSettings.FlashMode = photoOutput.SupportedFlashModes.Contains(flashMode) ? flashMode : photoOutput.SupportedFlashModes.First();
+		var photoOutputConnection = photoOutput.ConnectionFromMediaType(AVMediaTypes.Video.GetConstant()!);
+		photoOutputConnection!.VideoOrientation = videoOrientation;
```
Comment (bijington) [inline src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs:200]: Can you please remove the null forgiving operator `!` and check for `null` instead.

File: src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs [hunk 3]
```diff
@@ -241,11 +248,56 @@ protected virtual void Dispose(bool disposing)
...
 
+			orientationDidChangeObserver?.Dispose();
+			orientationDidChangeObserver = null;
+
 			photoOutput?.Dispose();
 			photoOutput = null;
 		}
 	}
 
+	// TODO: Move GetVideoOrientation() to Extensions class?
```
Comment (bijington) [inline src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs:259]: I think it's fine here if we don't need it anywhere else. Can we remove the comment if you don't want to move it?

File: src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs [hunk 4]
```diff
@@ -191,42 +196,44 @@ protected virtual async partial ValueTask PlatformTakePicture(CancellationToken
...
 
-		var dataBytes = ArrayPool<byte>.Shared.Rent((int)data.Length);
-
+		Stream? imageData;
 		try
 		{
-			Marshal.Copy(data.Bytes, dataBytes, 0, (int)data.Length);
-
-			cameraView.OnMediaCaptured(new MemoryStream(dataBytes));
+			imageData = result.Photo.FileDataRepresentation?.AsStream();
```
Comment (bijington) [inline src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs:222]: Will this result in more memory allocation vs the original ArrayPool.Rent approach?

**Review by bijington — CHANGES_REQUESTED** [review]
@Forestbrook Thank you for this. I have added some comments that hopefully you can address but I think the general changes look good

File: src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs (hunk 2 above)
Comment (Forestbrook) [reply src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs]: Done

File: src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs (hunk 3 above)
Comment (Forestbrook) [reply src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs]: Done

File: src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs [hunk 5]
```diff
@@ -191,42 +196,51 @@ protected virtual async partial ValueTask PlatformTakePicture(CancellationToken
...
 		}
-		catch (Exception ex)
+		catch
```
Comment (bijington) [inline src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs:231]: Would there be any value in including the exception details in the `OnMediaCaptureFailed` call?

**Review by bijington — COMMENTED** [review]
I haven't tested this myself but the overall changes look good

**Comment by bijington** [comment]
@tschbc thanks for confirming. I hadn't managed to test this so knowing someone else has really helps. I don't think this will merge before .NET 9.0 support as we are trying to get that in right now. But it'll go in shortly after

**Review by bijington — APPROVED** [review]
Thanks again for this!

**Comment by klindberg-magaya** [comment]
@bijington Hey Shawn I have been following this PR and it's of interest to our needs as well. It seems that you approved it, but maybe only SDKs were out of date. What's the process (maybe a time/date?) to getting released.
https://github.com/CommunityToolkit/Maui/pull/2312

NOTE I am currently getting build errors on the main branch as well as that pr branch. 
error CS0246: The type or namespace name 'MauiUIApplicationDelegate' could not be found +

**Comment by bijington** [comment]
@klindberg-magaya we are still trying to get the .NET 9.0 PR merged and then get all of these other bits in.

**Review by brminnick — APPROVED** [review]
Thanks @Forestbrook! Approved ✅

FYI - I removed the checks for `UIDevice.CurrentDevice.CheckSystemVersion(11, 0)` and `UIDevice.CurrentDevice.CheckSystemVersion(13, 0)` since `CommunityToolkit.Maui.Camera` only supports iOS 15+: https://github.com/CommunityToolkit/Maui/blob/e97b4a631ff9e3fc592bf1e3858beb3e32bc7938/src/CommunityToolkit.Maui.Camera/CommunityToolkit.Maui.Camera.csproj#L13

**Pull request merged**